src/plugins/ui_feature/format.ts

```diff
@@ -24,43 +24,43 @@ export class FormatPlugin extends UIPlugin {
    * The change of the decimal quantity is done one by one, the sign of the step
    * variable indicates whether we are increasing or decreasing.
    *
-   * If several cells are in the zone, the format resulting from the change of the
-   * first cell (with number type) will be applied to the whole zone.
+   * If several cells are in the zone, each cell's format will be individually
+   * evaluated and updated with the number type.
    */
   private setDecimal(sheetId: UID, zones: Zone[], step: SetDecimalStep) {
-    // Find the first cell with a number value and get the format
-    const numberFormat = this.searchNumberFormat(sheetId, zones);
-    if (numberFormat !== undefined) {
-      // Depending on the step sign, increase or decrease the decimal representation
-      // of the format
-      const newFormat = changeDecimalPlaces(numberFormat, step);
-      // Apply the new format on the whole zone
-      this.dispatch("SET_FORMATTING", {
-        sheetId,
-        target: zones,
-        format: newFormat,
-      });
-    }
+    // Find the each cell with a number value and get the format
+    zones.forEach((zone) => {
```

Comment:
```suggestion
    for(const zone of zones) {
```
We typycally use for loop and not forEach  in this repo as a convention